engine/runtime/src/main/java/org/enso/interpreter/node/callable/FunctionCallInstrumentationNode.java

```diff
@@ -85,23 +113,31 @@ boolean isExecutable() {
     }
 
     /**
-     * Handles execution through the polyglot API. Does not use the arguments provided, since the
-     * function arguments are already a part of this object.
+     * Handles execution through the polyglot API. Merges provide arguments with the ones
+     * that are already a part of this object.
      */
     @ExportMessage
     static class Execute {
       @Specialization
       static Object callCached(
-          FunctionCall functionCall,
-          Object[] arguments,
-          @Cached InteropApplicationNode interopApplicationNode) {
-        Object[] callArguments =
-            Arrays.copyOf(
-                functionCall.getArguments(), functionCall.getArguments().length + arguments.length);
-        System.arraycopy(
-            arguments, 0, callArguments, functionCall.getArguments().length, arguments.length);
-        return interopApplicationNode.execute(
-            functionCall.function, functionCall.state, callArguments);
+        FunctionCall functionCall,
+        Object[] arguments,
+        @Cached InteropApplicationNode interopApplicationNode,
+        @CachedLibrary(limit = "10") DynamicObjectLibrary objects
+      ) {
+        var args = Arrays.copyOf(
+          functionCall.getArguments(), functionCall.getArguments().length + arguments.length
+        );
+        System.arraycopy(arguments, 0, args, functionCall.getArguments().length, arguments.length);
+
+        var state = functionCall.state;
+        var old = objects.getOrDefault(state.getContainer(), 43L, null);
```

Comment:
I think this should be a constant somewhere in the source code, so that we can easily find all of its usages.

Reply:
Probably a unique key for the state. Create a constant for that?

Reply:
The benefit of `43L` is:
- it is a valid Enso value
- anyone can type it without worrying about (compile or runtime) dependencies

E.g. `43L` is good enough for prototyping, but yes, we need a real constant instead of `43L`:
- where to put it?
- how to make it accessible from all the places we needed it?
- how to make it unique (unlike `43L` which anyone can use)?
- how to make it inaccessible from other places?

Reply:
For it appears to be used only in `runtime` projects so I think it's safe to just save it somewhere there.

Reply:
I removed all the `State` manipulation in favor of - #7115